rasa/shared/core/generator.py

```diff
@@ -651,6 +651,15 @@ def _process_step(
 
         end_trackers = []
         for event in events:
+            if (
```

Comment:
Is this method only called from `rasa test`?

Reply:
No, it is also called during training and validation. But in those cases the action_texts would have been read from the training stories, this this `if` doesn't get triggered.